### PR TITLE
Add avro writing capabilities

### PR DESCRIPTION
**Pull request by achad4**
-- Due to the high memory requirements of transforming Avro - Parquet, I added the ability to write out Avro files directly

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -430,6 +430,18 @@
                             </descriptors>
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>create-jar-with-deps</id>
```
Comment (HenryCaiHaiying) [inline pom.xml]: That does this create-jar-with-deps do?  Will it make one big gigantic jar?

Comment (achad4) [reply pom.xml]: I can remove it (I use it for our setup at Sailthru) if you want. It creates an uber jar with all the dependencies

Comment (norrs) [reply pom.xml]: If you want to make a fatjar, isn't using maven-shadow-plugin the way to go? 

something like:

```
         <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-shade-plugin</artifactId>
                <configuration>
                    <!-- DO NOT MINIMIZE JARS -->
                    <minimizeJar>false</minimizeJar>
                    <shadedArtifactAttached>false</shadedArtifactAttached>
                    <createDependencyReducedPom>false</createDependencyReducedPom>
                    <keepDependenciesWithProvidedScope>false</keepDependenciesWithProvidedScope>
                    <useBaseVersion>true</useBaseVersion>
                    <filters>
                        <filter>
                            <artifact>*:*</artifact>
                            <excludes>
                                <!--
                                  I have no idea why adding HTML to main jar files have
                                  any benefit. But it seems pretty common.
                                 -->
                                <exclude>**/package.html</exclude>
                                <exclude>**/overview.html</exclude>
                                <exclude>about.html</exclude>
                                <!--
                                  The generated package-info.class is usually *not* included
                                  in compiled jars, but some projects have it in their jar file.
                                  It is never needed, and cannot be loaded as a class anyway.
                                 -->
                                <exclude>**/package-info.class</exclude>
                            </excludes>
                        </filter>
                    </filters>
                </configuration>
                <executions>
                    <!-- Run shade goal on package phase -->
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>shade</goal>
                        </goals>
                        <configuration>
                            <finalName>${project.artifactId}</finalName>
                            <transformers>
                                <!-- add Main-Class to manifest file -->
                                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                    <mainClass>${main.class}</mainClass>
                                </transformer>
                            </transformers>
                        </configuration>
                    </execution>
                </executions>
            </plugin>

```

https://maven.apache.org/plugins/maven-shade-plugin/ & https://maven.apache.org/plugins/maven-shade-plugin/usage.html

This code above is just copied and slightly modified from what we've used internally, so need to be adjusted to be working with this project.

Comment (HenryCaiHaiying) [reply pom.xml]: I prefer not having a fatjar by default.

**Comment by achad4** [comment]
Hey @HenryCaiHaiying I removed the "create-jar-with-deps" step and fixed the failing tests

**Comment by HenryCaiHaiying** [comment]
Looks good to me, thanks for the contribution.

**Pull request merged**